modules/ts/include/opencv2/ts/ts_gtest.h

```diff
@@ -17478,6 +17478,277 @@ class CartesianProductGenerator10
 };  // class CartesianProductGenerator10
 
 
+template <typename T1, typename T2, typename T3, typename T4, typename T5,
```

Comment:
I hope it's OK that I put additional classes in that file

Reply:
I think it's a bad idea :)

Reply:
Lets to discuss it locally

Reply:
Instead you could just wrap parameters of your kernel to a structure and instantiate the test with two-three versions of this structure.

@alalek will also mention to replace this structure by enum, probably a good way to go as well.

Reply:
number of parameters was decreased according to the tests in imgproc